apps/condo/domains/miniapp/schema/B2BAppAccessRightSet.test.js

```diff
@@ -0,0 +1,650 @@
+        await updateTestB2BAppAccessRightSet(admin, accessRightSet.id, {
+            canReadOrganizations: true,
+        })
+
+        // you cannot update a field "canManageOrganizations". Only reading! Always false!
+        await catchErrorFrom(async () => {
+            await updateTestB2BAppAccessRightSet(admin, accessRightSet.id, {
+                canManageOrganizations: false,
+            })
+        }, (e) => {
+            expect(e.errors[0].message).toContain('Field "canManageOrganizations" is not defined by type "B2BAppAccessRightUpdateInput"')
+        })
+
+        // B2BApp with permission "canReadOrganization"
+        const countWithPermissions = await Organization.count(serviceUserClient, {})
+        expect(countWithPermissions).toBe(1)
+        const organizationWithPermissions = await Organization.getOne(serviceUserClient, { id: organization.id })
+        expect(organizationWithPermissions.id).toBe(organization.id)
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestOrganization(serviceUserClient)
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await updateTestOrganization(serviceUserClient, organization.id, {})
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await Organization.softDelete(serviceUserClient, organization.id)
+        })
+    })
+
+    test('Property', async () => {
+        const [organization] = await registerNewOrganization(user)
+
+        const [app] = await createTestB2BApp(admin)
+        await createTestB2BAppContext(admin, app, organization, { status: 'Finished' })
+        const [right] = await createTestB2BAppAccessRight(admin, serviceUser, app)
+
+        const [property] = await createTestProperty(user, organization)
+
+        // B2BApp without permissions
+        const countWithoutPermissions = await Property.count(serviceUserClient, {})
+        expect(countWithoutPermissions).toBe(0)
+        const propertyWithoutPermissions = await Property.getOne(serviceUserClient, { id: property.id })
+        expect(propertyWithoutPermissions).toBeUndefined()
+
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await createTestProperty(serviceUserClient, organization)
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await updateTestProperty(serviceUserClient, property.id, {})
+        })
+        await expectToThrowAccessDeniedErrorToObj(async () => {
+            await Property.softDelete(serviceUserClient, property.id)
+        })
+
+        const countByUser = await Property.count(user, {})
+        expect(countByUser).toBe(1)
+        const propertyByUser = await Property.getOne(user, { id: property.id })
+        expect(propertyByUser.id).toBe(property.id)
+
+        // add permissions for B2BApp
+        const [accessRightSet] = await createTestB2BAppAccessRightSet(admin, app, {
```

Comment:
How about separate read / manage checks?

For example:
I can update property area without having read permission to organization.
I can read property addresses without having access to org and so on.

Reply:
Also I've found an issue / question:

Do we allowing apps to change property address?

Reply:
> Do we allowing apps to change property address?

In the plugin, at the moment, the accesses for fields do not change in any way. Only accesses to schemas change.

That is, default checks are used for fields, which are specified in schemas

If this is done now, it will complicate current plugin logic and its configuration. Therefore, I suggest leaving it for the next iteration)

Reply:
> How about separate read / manage checks?

Do you want to write this for all schemas? This is madness! :)